### ISSUE TITLE
is_parent not working correctly

### ISSUE THREAD

**Issue by AndersMFJ** [statamic/cms#2298]
## Bug Description
I'm using is_parent to build a navigation with an active state, similar to the example in the documentation ([Nav tag](https://statamic.dev/tags/nav)). The navigation is meant to show only those parts of the hierarchy that are current or parents of current. It's been working fine until recently, when I upgraded to 3.0. The code below has been simplified for brevity by removing styles.  

```
<ul>
 {{ nav }}
    <li>
        <a href="{{ url }}">{{ title }}</a>
        {{ if children }}
            <ul class="{{ is_current || is_parent ? '' : 'hidden' }}">{{ *recursive children* }}</ul>
        {{ /if }}
    </li>
{{ /nav }}
</ul>
```

This gets me:
![Skärmavbild 2020-08-26 kl  11 30 29](https://user-images.githubusercontent.com/50993202/91287137-9e46c880-e78f-11ea-8413-43ead6c13c9f.png)

What I want is this:
![Skärmavbild 2020-08-26 kl  11 31 36](https://user-images.githubusercontent.com/50993202/91287224-b9193d00-e78f-11ea-95e9-4a5347da6d23.png)

## How to Reproduce
Use the example from the documentation, no parent <a> tag gets the "on" class. 
## Environment
Statamic 3.0.0 Pro
Laravel 7.26.0
PHP 7.4.6
edalzell/rss 1.0.2
jonassiewertsen/statamic-oh-dear 1.2.1
statamic/collaboration dev-master
statamic/seo-pro 2.0.10

**Install method (choose one):**
- Fresh install from `statamic/statamic`

**Comment by jasonvarga** [comment]
Can you show what's in your config/statamic/sites.php?

**Comment by AndersMFJ** [comment]
```
    'sites' => [

        'sv' => [
            'name' => 'Svenska',
            'locale' => 'sv_SE',
            'url' => '/',
        ],

        'en' => [
            'name' => 'English',
            'locale' => 'en_US',
            'url' => '/en/',
        ]

    ]
```

**Comment by mikemartin** [comment]
May not be related, but it seems like `is_parent` will only work for my mounted collections if the `mount` segment is in the route.

For example, I have a blog collection where I'd prefer the route to be `/{slug}` but then on my blog post entry `is_parent` returns false. It's only when I change the collection route to `/{mount}/{slug}` that `is_parent` returns true.

**Comment by AndersMFJ** [comment]
I'm thinking this is a multisite issue because I have another site which is not multisite and the same code works perfectly there. @jasonvarga

**Comment by AndersMFJ** [comment]
Here's another weird clue: It works correctly on the first subentry of each category. So this page works: [https://mfj.se/lar-dig-mer/vald](https://mfj.se/lar-dig-mer/vald) but this doesn't [https://mfj.se/lar-dig-mer/vald-gar-att-foerebygga](https://mfj.se/lar-dig-mer/vald-gar-att-foerebygga)

**Comment by AndreasSchantl** [comment]
@AndersMFJ Are you using ``redirect @child`` at any chance? If so, check out #2359

**Comment by AndersMFJ** [comment]
@AndreasSchantl Thank you, that fixed my issue!

**Comment by AndreasSchantl** [comment]
Great to hear. Waiting for it to be merged!

**Comment by jasonvarga** [comment]
Andreas' fix was included in #3522
